agent/agent.go

```diff
@@ -389,8 +389,8 @@ func (a *Agent) UpdateTaskStatus(ctx context.Context, taskID string, status *api
 func (a *Agent) nodeDescriptionWithHostname(ctx context.Context) (*api.NodeDescription, error) {
 	desc, err := a.config.Executor.Describe(ctx)
 
-	// Override hostname
-	if a.config.Hostname != "" && desc != nil {
+	// Only override hostname if it is not received from the call to Describe
+	if desc != nil && desc.Hostname == "" {
```

Comment:
Never mind, I misunderstood. But yes, I think the check would be redundant. If `desc.Hostname == ""`, then it doesn't matter whether `a.config.Hostname` is empty or not.